### ISSUE TITLE
optionally specify an image

### ISSUE THREAD

**Issue by hayd** [frewsxcv/rust-notifica#3]
I'm not sure what a cross platform API could look like for this.

 it could be an enum rather than arbitrary images...

**Comment by kunicmarko20** [comment]
https://github.com/hoodie/notify-rust/blob/940dbcb112a9de571dfaca20bb25eb10d9cf067d/src/lib.rs#L264-L278

notify-rust allows data, path or a system image (if I am correct)?

Not sure about other libs but if this would be possible cross platform it would be awesome

**Comment by flying-sheep** [comment]
- [On Linux](https://developer.gnome.org/notification-spec/#icons-and-images), you specify an optional icon and an optional image. Both can be specified as `file://...` URIs or as raw image data. You can also specify a standard name for the icon to use a native theme icon.
- On macOS, you specify an app bundle identifier, so you can use any app’s icon, I guess. One can get the bundle identifier from an app name using `mac_notification_sys::get_bundle_identifier("Some App Name")`

    `mac_notification_sys` doesn’t support adding images by path or data, but the [new, non-deprecated API](https://developer.apple.com/documentation/usernotifications/unmutablenotificationcontent?language=objc) allows to add images as “attachments”.

- [On Windows](https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/adaptive-interactive-toasts), you create XML to specify the notification. By default the app logo is used but you can do

    ```xml
    <toast launch="app-defined-string">
      <visual>
        <binding template="ToastGeneric">
          <text>Sample</text>
          <text>This is a simple toast notification example</text>
          <image placement="AppLogoOverride" src="some-url-or-path.png" />
        </binding>
      </visual>
      ...
    </toast>
    ```

    There’s a *lot* more possible on windows, including big banner-style images.

So in summary, the user should get the ability to specify both an icon and an image.

- The icon can have

    1. a linux theme name
    2. a mac bundle ID
    3. a path or image data (used on windows and on linux if you didn’t specify a linux theme name)

- The image can have a path or image data and is only for linux and windows (unless we switch from `mac_notification_sys` to something using the new macOS API)

**Comment by kunicmarko20** [comment]
So for macOS there is no way to provide image path? Only existing app icons?

**Comment by flying-sheep** [comment]
see edit
